InnoClinic.ProfilesAPI/Infrastructure/Repositories/BaseRepository.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Domain.Interfaces;
+using Infrastructure.Context;
+using Microsoft.EntityFrameworkCore;
+
+namespace Infrastructure.Repositories;
+
+public class BaseRepository<T> : IRepository<T> where T : class
+{
+    protected ApplicationDbContext context;
+
+    public BaseRepository(ApplicationDbContext context)
+    {
+        this.context = context;
+    }
+
+    public async Task<T> CreateAsync(T entity, CancellationToken cancellationToken = default)
+    {
+        await context.Set<T>().AddAsync(entity, cancellationToken);
+        await context.SaveChangesAsync(cancellationToken);
+        return entity;
+    }
+
+    public async Task DeleteAsync(T entity, CancellationToken cancellationToken = default)
+    {
+        context.Set<T>().Remove(entity);
+        await context.SaveChangesAsync(cancellationToken);
+    }
+
+    public async Task<List<T>> GetAllAsync(CancellationToken cancellationToken = default) =>
+         await context.Set<T>().AsNoTracking().ToListAsync(cancellationToken);
+
+    public async Task<T?> GetByIdAsync(Guid id, CancellationToken cancellationToken = default) =>
+       await context.Set<T>().FirstOrDefaultAsync(entity => EF.Property<Guid>(entity, "Id") == id, cancellationToken);
```

Comment:
here you can also add AsNoTraking, but you need to add param for method (bool trackChanges, for example)